contracts/VersionResolver.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity ^0.4.0;
+
+contract VersionResolver {
+    address owner;
+    mapping(string=>address) addresses;
+
+    modifier only_owner() {
+        if(msg.sender != owner) throw;
+        _;
+    }
+
+    function VersionResolver() {
+        owner = msg.sender;
+    }
+
+    // Our resolver supports (supportsInterface & addr) interfaces
+    function supportsInterface(bytes4 interfaceID) constant returns (bool) {
+        return interfaceID == 0x3b3b57de
+            || interfaceID == 0x01ffc9a7;
```

Comment:
Anything that doesn't mutate state can be a view function and so gas free.

Reply:
Ah I copied from a tutorial but yes your right

Reply:
You can do it for more than that function e.g.  getAddress